l1-contracts/contracts/governance/PermanentRestriction.sol

```diff
@@ -22,10 +25,21 @@ import {IPermanentRestriction} from "./IPermanentRestriction.sol";
 /// properties are preserved forever.
 /// @dev To be deployed as a transparent upgradable proxy, owned by a trusted decentralized governance.
 /// @dev Once of the instances of such contract is to ensure that a ZkSyncHyperchain is a rollup forever.
-contract PermanentRestriction is IRestriction, IPermanentRestriction, Ownable2Step {
+contract PermanentRestriction is IRestriction, IPermanentRestriction, Ownable2StepUpgradeable {
+    /// @notice We use try-catch to test whether some of the conditions should be checked.
+    /// To avoid attacks based on teh 63/64 gas limitations, we ensure that each such call 
+    /// has at least this amount.
+    uint256 constant MIN_GAS_FOR_FALLABLE_CALL = 5_000_000; 
```

Comment:
```suggestion
    uint256 internal constant MIN_GAS_FOR_FALLABLE_CALL = 5_000_000; 
```

Reply:
moved to out of the contract to import in tests